backend/app/routes.py

```diff
@@ -117,42 +117,41 @@ def get_links_between_two_nodes(from_node_id, to_node_id):
 
     with connection:
         with connection.cursor() as cursor:
-            #Uses pg_routing to route between the start node and end node on the HERE
-            #links network. Returns inputs and an array of link_dirs and a unioned line
-            #TODO: convert to here_gis.get_links_between_nodes
+            #Uses here_gis.get_links_btwn_nodes_22_2 to route between the start node and end node on the HERE
+            #links network. Returns an array of link objects.
+            #TODO: create unique pathnames
             select_sql = '''
                 WITH results as (
                     SELECT *
-                    FROM pgr_dijkstra(
-                        'SELECT id, source::int, target::int, length::int AS cost FROM here.routing_streets_name',
-                        %(node_start)s,
-                        %(node_end)s
-                    )
+                    FROM here_gis.get_links_btwn_nodes_22_2(
+                        '%(node_start)s',
+                        '%(node_end)s'
+                    ),
+                    UNNEST (links) WITH ORDINALITY AS unnested (link_dir, seq)
                 )
 
                 SELECT 
-                    %(node_start)s,
-                    %(node_end)s,
-                    array_agg(st_name),
-                    array_agg(link_dir),
-                    ST_AsGeoJSON(ST_union(ST_linemerge(geom))) AS geometry
-                FROM results
-                INNER JOIN here.routing_streets_name on edge = id'''
+					results.link_dir,
+					results.seq,
+					seg_lookup.segment_id,
+					ST_AsGeoJSON(streets.geom) AS geom,
+					ST_Length( ST_Transform(streets.geom,2952) ) / 1000 AS length_km
+				FROM results
+				JOIN here.routing_streets_22_2 AS streets USING ( link_dir )
+				JOIN congestion.network_links_22_2 AS seg_lookup USING ( link_dir )
+				ORDER BY seq;

```

Comment:
Indented with tabs. Please use spaces.

Reply:
Found the problem - when I copypasted back from pgadmin, it kept tab chars instead of spaces!

Reply:
You may also want to configure the tabbing/spacing options in PGAdmin. 

File -> Preferences -> Query Tool -> SQL Formatting